### PR TITLE
[v0.33] feat / refactor reporting proxy to send order event logs to remote log collector

### PR DESCRIPTION
**Pull request by michaelc9**
**Before submitting this PR, please make sure**:
- [ x] Your code builds clean without any errors or warnings
- [ x] You are using approved title ("feat/", "fix/", "docs/", "refactor/", etc)

**Optional**:
- Related Github issue:
- Related Clubhouse Story:
https://app.clubhouse.io/coinalpha/story/10164/add-datadog-hummingbot-logs-analysis-to-kucoin-environment-verification-tool

**A description of the changes proposed in the pull request**:

### CODE REVIEW

File: hummingbot/logger/reporting_proxy_handler.py [hunk 1]
```diff
@@ -116,6 +150,9 @@ def flush(self, send_all=False):
                 if len(self._log_queue) > min_send_capacity:
                     self.send_logs(self._log_queue)
                     self._log_queue = []
+            if len(self._event_queue) > min_send_capacity:
+                self.send_event(self._event_queue)
+                self._event_queue = []
```
Comment (pw96pw96pw96) [inline hummingbot/logger/reporting_proxy_handler.py:155]: Do line 154 and 155 together need to be protected by a lock? :)

**Review by Nullably — APPROVED** [review]
Pending on the question by @pw96pw96pw96 
The rest of the code looks good to me.

File: hummingbot/logger/reporting_proxy_handler.py (hunk 1 above)
Comment (michaelc9) [reply hummingbot/logger/reporting_proxy_handler.py:167]: The `logging.Handler` base class manages a thread lock internally and the lock is acquired at the beginning of the flush function.

File: hummingbot/core/event/events.py [hunk 2]
```diff
@@ -154,9 +155,11 @@ class BuyOrderCompletedEvent(NamedTuple):
     quote_asset_amount: Decimal
     fee_amount: Decimal
     order_type: OrderType
+    exchange_order_id: Optional[str] = ""
```
Comment (Nullably) [inline hummingbot/core/event/events.py:158]: Shouldn't `Optional[str]` be assigned None value initially?

**Review by Nullably — CHANGES_REQUESTED** [review]

File: hummingbot/core/event/events.py (hunk 2 above)
Comment (michaelc9) [reply hummingbot/core/event/events.py]: Right. Refactored.

**Review by Nullably — APPROVED** [review]
LGTM

**Pull request merged**